### PR TITLE
Compose expert contributed to login view model

### PR DESCRIPTION
**Pull request by mkZ3R0**

### CODE REVIEW

File: app/src/main/java/app/kitabcha/presentation/LoginScreen.kt [hunk 1]
```diff
@@ -57,6 +57,7 @@ fun Content(
     ) {
         val username by loginViewModel.userName.collectAsStateWithLifecycle()
         val password by loginViewModel.userPassword.collectAsStateWithLifecycle()
+        val alreadyexists by loginViewModel.alreadyExists.collectAsStateWithLifecycle()
```
Comment (Mubashir22009) [inline app/src/main/java/app/kitabcha/presentation/LoginScreen.kt:60]:
```suggestion
        val alreadyexists by loginViewModel.alreadyExists.collectAsStateWithLifecycle()
        
```

File: app/src/main/java/app/kitabcha/presentation/LoginScreenViewModel.kt [hunk 2]
```diff
@@ -21,7 +21,10 @@ class LoginScreenViewModel
         private val libraryRepository: LibraryRepository,
     ) : ViewModel() {
         fun insertUser(user: UserEntity) {
+            var totalBefore: Int
+            var totalNow: Int
```
Comment (Mubashir22009) [inline app/src/main/java/app/kitabcha/presentation/LoginScreenViewModel.kt]:
```suggestion
```

File: app/src/main/java/app/kitabcha/presentation/LoginScreenViewModel.kt [hunk 3]
```diff
@@ -21,7 +21,10 @@ class LoginScreenViewModel
...
             viewModelScope.launch(IO) {
+                totalBefore = repository.getTotalUserCount()
```
Comment (Mubashir22009) [inline app/src/main/java/app/kitabcha/presentation/LoginScreenViewModel.kt]:
```suggestion
                val totalBefore = repository.getTotalUserCount()
```

File: app/src/main/java/app/kitabcha/presentation/LoginScreenViewModel.kt [hunk 4]
```diff
@@ -31,6 +34,10 @@ class LoginScreenViewModel
                         ),
                     )
                 }
+                totalNow = repository.getTotalUserCount()
```
Comment (Mubashir22009) [inline app/src/main/java/app/kitabcha/presentation/LoginScreenViewModel.kt]:
```suggestion
                val totalNow = repository.getTotalUserCount()
```

File: app/src/main/java/app/kitabcha/presentation/LoginScreen.kt [hunk 5]
```diff
@@ -152,5 +153,10 @@ fun Content(
                 }
             }
         }
+        if (alreadyexists) {
+            Toast.makeText(localContext, "Username Taken Like Your Waifu", Toast.LENGTH_SHORT)
```
Comment (Mubashir22009) [inline app/src/main/java/app/kitabcha/presentation/LoginScreen.kt:161]: no

**Review by Mubashir22009 — CHANGES_REQUESTED** [review]

**Review by mkZ3R0 — COMMENTED** [review]
aye phar

**Pull request merged**